parley/src/style/styleset.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2024 the Parley Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use core::mem::Discriminant;
+use hashbrown::HashMap;
+
+type StyleProperty<Brush> = crate::StyleProperty<'static, Brush>;
```

Comment:
I guess? Having to implement the conversion back from `TextStyle<'static, Brush>` to an iterator of StyleProperties is quite miserable though, isn't it?

Reply:
I was rather thinking of adding a method to `RangedBuilder` that takes a `TextStyle` directly. `TreeBuilder` already has one, so it could be copied from there.

Reply:
I suggested something similar to `TextStyle` elsewhere, where each property is wrapped in an `Option` so we know whether the property is explicitly set or not, but the same concerns around ergonomics remain.